### ISSUE TITLE
Dotnet - Attempt to save a screenshot to a directory which does not exist throw a cryptic generic GDI error

### ISSUE THREAD

**Issue by sandorfr** [SeleniumHQ/selenium#4645]
## Meta -
OS:  
Windows
Selenium Version:  
3.5.1
Browser:  
any

Browser Version:  
any

When we try to save a screenshot to a directory which does not exist, It throws with the following error:
```
A generic error occurred in GDI+
```

## Expected Behavior -
It should either create the directory or throw an explicit `DirectoryNotFoundException`

## Actual Behavior -
It throws a cryptic generic GDI exception

## Steps to reproduce -

Add the following to an existing test
```csharp

                var testScreenShotPath = "C:\\PathToADirectoryWhichDoesNotExists";                
                var path = Path.Combine(testScreenShotPath, ${Guid.NewGuid()}.jpg");
                var screenshot = ((ITakesScreenshot)BrowserInstance).GetScreenshot();
                screenshot.SaveAsFile(path, ScreenshotImageFormat.Jpeg);
```

Run the test

Additional reference can be found [on Stackoverflow](https://stackoverflow.com/questions/29973934/a-generic-error-occurred-in-gdi-screenshot-saveasfile-externalexception)

**Comment by sandorfr** [comment]
I've submitted a PR with a fix proposal throwing a DirectoryNotFoundException

**Comment by jimevans** [comment]
Fixed in 96c1b99.
